packages/firestore/src/lite-api/aggregate.ts

```diff
@@ -47,10 +52,130 @@ import { LiteUserDataWriter } from './reference_impl';
+  query: Query<unknown>,
+  aggregateSpec: T
+): Promise<AggregateQuerySnapshot<T>> {
   const firestore = cast(query.firestore, Firestore);
   const datastore = getDatastore(firestore);
+
+  const internalAggregates = mapToArray(aggregateSpec, (aggregate, alias) => {
+    // TODO (sum/avg) should alias validation be performed or should that be
+    // delegated to the backend?
+
+    return new AggregateImpl(
+      alias,
+      aggregate._aggregateType,
+      aggregate._internalFieldPath
+    );
+  });
+
+  // Run the aggregation and convert the results
+  return invokeRunAggregationQueryRpc(
+    datastore,
+    query._query,
+    internalAggregates
+  ).then(aggregateResult =>
+    convertToAggregateQuerySnapshot(firestore, query, aggregateResult)
+  );
+}
+
+function convertToAggregateQuerySnapshot<T extends AggregateSpec>(
+  firestore: Firestore,
+  query: Query<unknown>,
+  aggregateResult: ObjectValue
+): AggregateQuerySnapshot<T> {
   const userDataWriter = new LiteUserDataWriter(firestore);
-  return new CountQueryRunner(query, datastore, userDataWriter).run();
+  const querySnapshot = new AggregateQuerySnapshot<T>(
+    query,
+    userDataWriter,
+    aggregateResult
+  );
+  return querySnapshot;
+}
+
+/**
+ * Create an AggregateField object that can be used to compute the sum of
+ * a specified field over a range of documents in the result set of a query.
+ * @param field Specifies the field to sum across the result set.
+ * @internal TODO (sum/avg) remove when public
+ */
+export function sum(field: string | FieldPath): AggregateField<number> {
+  return new AggregateField('sum', fieldPathFromArgument('sum', field));
+}
+
+/**
+ * Create an AggregateField object that can be used to compute the average of
+ * a specified field over a range of documents in the result set of a query.
+ * @param field Specifies the field to average across the result set.
+ * @internal TODO (sum/avg) remove when public
+ */
+export function average(
+  field: string | FieldPath
+): AggregateField<number | null> {
+  return new AggregateField('avg', fieldPathFromArgument('average', field));
+}
+
+/**
+ * Create an AggregateField object that can be used to compute the count of
+ * documents in the result set of a query.
+ * @internal TODO (sum/avg) remove when public
+ */
+export function count(): AggregateField<number> {
+  return new AggregateField('count');
+}
+
+/**
+ * Compares two 'AggregateField` instances for equality.
+ *
+ * @param left Compare this AggregateField to the `right`.
+ * @param right Compare this AggregateField to the `left`.
+ */
+export function aggregateFieldEqual(
+  left: AggregateField<unknown>,
+  right: AggregateField<unknown>
+): boolean {
+  return (
+    left instanceof AggregateField &&
+    right instanceof AggregateField &&
+    left._internalFieldPath?.canonicalString() ===
+      right._internalFieldPath?.canonicalString() &&
+    left._aggregateType === right._aggregateType
```

Comment:
ultra nit: it's better to check `left._aggregateType === right._aggregateType` _before_ checking the canonicalStrings. It'd be infinitesimally more efficient for cases when the aggregate types don't match

Reply:
Yeah, it's cleaner. Fixed.